### PR TITLE
syncpoint (ticdc): support external ts read for TiCDC's downstream when syncpoint is enabled.

### LINKED ISSUE #7419: support external ts read for TiCDC's downstream when syncpoint is enabled.
### Is your feature request related to a problem?

as title

### Describe the feature you'd like

as title

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #7419 

### What is changed and how it works?
Set downstream system variable `tidb_external_ts` as `secondary ts` after we insert a new `ts-map` into downstream's syncpoint table.

**Note**: If the downstream TiDB does not support `tidb_external_ts`, changefeed will keep silent.

#### Tests 

 - Manual test (add detailed scripts or steps below)
1. create a changefeed and enable syncpoint.
2. use external ts read (session level) in downstrem during ticdc replicating, works well.

### Release note 

```release-note
Support external ts read for TiCDC's downstream when syncpoint is enabled.
```

### CODE REVIEW

**Comment by lance6716** [comment]
please note that `If the `tidb_enable_external_ts_read` is on, the tidb will be in read-only mode`, so CDC need some trick (maybe some special privilege granted to the user) to write data to TiDB

**Comment by lance6716** [comment]
Also will there be multiple changefeed write to one downstream cluster? There's only one `tidb_external_ts` for it

File: cdc/sink/mysql/mysql_syncpoint_store.go [hunk 1]
```diff
@@ -231,6 +232,22 @@ func (s *mysqlSyncPointStore) SinkSyncPoint(ctx context.Context,
 		return cerror.WrapError(cerror.ErrMySQLTxnError, err)
 	}
 
+	// set global tidb_external_ts to secondary ts
```
Comment (nongfushanquan) [inline cdc/sink/mysql/mysql_syncpoint_store.go:235]: Pls add some comments about the tidb's version which supports this variables

**Comment by YangKeao** [comment]
> please note that `` If the `tidb_enable_external_ts_read` is on, the tidb will be in read-only mode ``, so CDC need some trick (maybe some special privilege granted to the user) to write data to TiDB

`RESTRICTED_REPLICA_WRITER_ADMIN` ?

**Comment by YangKeao** [comment]
> Also will there be multiple changefeed write to one downstream cluster? There's only one `tidb_external_ts` for it

@songrijie PTAL. It seems that the current design of `tidb_external_ts` doesn't support multiple changefeed sharing the same tidb cluster, as this variable is global, and has only one. Technically, `PD` could store multiple timestamps in different slots, but the user interface of TiDB should be redesigned to be able to set and use different external timestamps :thinking:

File: cdc/sink/mysql/mysql_syncpoint_store.go (hunk 1 above)
Comment (asddongmen) [reply cdc/sink/mysql/mysql_syncpoint_store.go:235]: I will add it later.

**Review by nongfushanquan — APPROVED** [review]

File: cdc/sink/mysql/mysql_syncpoint_store.go [hunk 2]
```diff
@@ -17,12 +17,13 @@ import (
 	"context"
 	"database/sql"
 	"fmt"
+	"github.com/pingcap/tiflow/pkg/errorutil"
```
Comment (hi-rustin) [inline cdc/sink/mysql/mysql_syncpoint_store.go:20]: Please fmt this.

File: cdc/sink/mysql/mysql_syncpoint_store.go [hunk 3]
```diff
@@ -231,6 +232,23 @@ func (s *mysqlSyncPointStore) SinkSyncPoint(ctx context.Context,
...
+	if err != nil {
+		if errorutil.IsSyncPointIgnoreError(err) {
+			// TODO(dongmen): to confirm if we need to log this error.
+			log.Warn("set global external ts failed, ignore this error", zap.Error(err))
+		} else {
+			err2 := tx.Rollback()
+			if err2 != nil {
+				log.Error("failed to write syncpoint table", zap.Error(cerror.WrapError(cerror.ErrMySQLTxnError, err2)))
+			}
+			return cerror.WrapError(cerror.ErrMySQLTxnError, err)
```
Comment (hi-rustin) [inline cdc/sink/mysql/mysql_syncpoint_store.go:248]: Can we just wrap it once?

File: pkg/errorutil/util.go [hunk 4]
```diff
@@ -135,3 +135,17 @@ func IsRetryableDDLError(err error) bool {
...
+
+func IsSyncPointIgnoreError(err error) bool {
+	err = errors.Cause(err)
+	mysqlErr, ok := err.(*gmysql.MySQLError)
+	if !ok {
+		return false
+	}
+	// We should ignore the error when the downstream has no
+	// such system variable for compatibility.
+	if mysqlErr.Number == mysql.ErrUnknownSystemVariable {
```
Comment (hi-rustin) [inline pkg/errorutil/util.go:147]: Just `return mysqlErr.Number == mysql.ErrUnknownSystemVariable` ?

File: pkg/errorutil/util_test.go [hunk 5]
```diff
@@ -111,3 +111,19 @@ func TestIsRetryableDDLError(t *testing.T) {
 		require.Equal(t, c.ret, IsRetryableDDLError(c.err))
 	}
 }
+
+func TestIsSyncPointIgnoreError(t *testing.T) {
+	cases := []struct {
```
Comment (hi-rustin) [inline pkg/errorutil/util_test.go:116]: t.Parallel

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**